app/src/main/java/com/erwanlbp/calculit/activity/MainActivity.java

```diff
@@ -71,8 +82,15 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             if (resultCode == ActivityCode.RC_BACK_HOME) {
                 // TODO Finished Play
                 User.getInstance().updateHighScore(gameConfig.getDifficulty(), gameConfig.getLevel() - 1); // -1 Cause it mean he failed this level
-                // Reset game config
-                gameConfig = new GameConfig(gameConfig.getDifficulty());
+                TextView textView = (TextView) findViewById(R.id.buttonLaunchGame);
+                if (data.getBooleanExtra(PrintResultsActivity.CORRECT_ANSWER, false)) {
+                    gameConfig = gameConfig.nextLevel();
+                    textView.setText(R.string.continue_game);
```

Comment:
Add a private updateUI() that take care of printing the right content in the button